htdocs/install/upgrade2.php

```diff
@@ -5137,3 +5144,62 @@ function migrate_export_import_profiles($mode = 'export')
 		print '<tr class="trforrunsql" style=""><td class="wordbreak" colspan="4">'.$langs->trans("NothingToDo")."</td></tr>\n";
 	}
 }
+
+/**
+ * Migrate Rank into contract  line
+ *
+ * @return  void
+ */
+function migrate_contractdet_rank()
+{
+
+	global $db, $langs;
+
+	$error = 0;
+	$resultstring = '';
+
+	$db->begin();
+	print '<tr class="trforrunsql"><td colspan="4">';
+
+	 $sql = 'SELECT c.rowid as cid ,cd.rowid as cdid,cd.rang FROM '.$db->prefix().'contratdet as cd INNER JOIN '.$db->prefix().'contrat as c ON c.rowid=cd.fk_contrat AND cd.rang=0';
+	 $sql .=' ORDER BY c.rowid,cd.rowid';
+
+	$resql = $db->query($sql);
+	if ($resql) {
+		$currentRank=0;
+		$current_contract=0;
+		while ($obj = $db->fetch_object($resql)) {
+			if (empty($current_contract) || $current_contract==$obj->cid) {
+				$currentRank++;
+			} else {
+				$currentRank=1;
+			}
+
+			$sqlUpd = 'UPDATE '.$db->prefix().'contratdet SET rang='.(int) $currentRank.' WHERE rowid='.(int) $obj->cdid;
```

Comment:
Same here, use " foe sql forging and ' when it is a sql string inside the php string